test/acl.test.js

```diff
@@ -472,3 +476,130 @@ describe('access check', function() {
       });
   });
 });
+
+describe('Authorized roles propagated in RemotingContext', function() {
+  var app, request, accessToken;
+  var models = {};
+
+  beforeEach(setupAppAndRequest);
+
+  it('contains all authorized roles for a principal if query is allowed', function() {
+    return createACLs('MyTestModel', [
+      {permission: ACL.ALLOW, principalId: '$everyone'},
+      {permission: ACL.ALLOW, principalId: '$authenticated'},
+      {permission: ACL.ALLOW, principalId: 'myRole'},
+    ])
+    .then(function() {
+      return request.get('/MyTestModels')
+        .set('X-Access-Token', accessToken.id)
+        .expect(200);
```

Comment:
This code is repeated in all tests, can you extract a shared helper function please? E.g.

```js
// usage
.then(makeAuthorizedHttpRequest)

// implementation
function makeAuthorizedHttpRequest() {
  return request.get('/MyTestModels')
    .set(...)
    .expect(200);
}
```